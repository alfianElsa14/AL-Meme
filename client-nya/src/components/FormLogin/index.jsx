import React, { useState } from 'react'
import { FormattedMessage } from 'react-intl';
import { useNavigate } from 'react-router-dom'
import { useDispatch } from 'react-redux'

import { googleLogin, loginUser } from '@containers/Client/actions';

import GoogleIcon from '@mui/icons-material/Google';
import classes from './style.module.scss'

function FormLogin() {
    const navigate = useNavigate()
    const dispatch = useDispatch()

    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch(loginUser({ email, password }));
    }

    const handleGoogleLogin = () => {
        dispatch(googleLogin());
    };
    return (
        <div
            className={classes.formContainer}
            data-testid="formContainer"
        >
            <h3><FormattedMessage id="app_login" /></h3>
            <div className={classes.form}>
                <form action="" onSubmit={handleLogin}
                    className={classes.formLogin}
                    data-testid="formLogin"
                >
                    <label htmlFor="">Email :</label>
                    <input
                        data-testid="email"
                        type="email"
                        value={email}
                        onChange={(e) => setEmail(e.target.value)} />
                    <label htmlFor="">Password :</label>
                    <input
                        data-testid="password"
                        type="password"
                        value={password}
                        onChange={(e) => setPassword(e.target.value)} />
                    <button type='submit'>
                        <FormattedMessage id="app_login" />
                    </button>
                </form>
                <button
                    data-testid="google"
                    onClick={handleGoogleLogin}
                    className={classes.googleButton}
                >
                    <GoogleIcon />
                    <FormattedMessage id="app_login_with_google" />
                </button>
            </div>
            <p>
                <FormattedMessage id="app_dont_have_account" />
                click <span
                    data-testid="toRegister"
                    onClick={() => navigate('/register')}
                >
                    <FormattedMessage id="app_here" />
                </span>
            </p>
        </div>
    )
}

export default FormLogin