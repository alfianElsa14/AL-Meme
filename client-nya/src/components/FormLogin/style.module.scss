@import '@styles/base';

.formContainer {
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-left: none;
    color: $color-white;
    background-color: $color-marine-blue;
    border-radius: 5px;

    @include apply-at-md-size {
        padding: 20px;
    }

    .formLogin {
        display: flex;
        flex-direction: column;
        width: 300px;
        color: $color-white;

        input {
            margin-bottom: 30px;
            height: 40px;
            border: none;
            border-radius: 5px;
        }

        label {
            font-size: 17px;
            margin-bottom: 10px;
        }

        button {
            border: none;
            height: 40px;
            border-radius: 5px;
            background-color: rgb(184, 179, 255);
            cursor: pointer;
        }

        button:hover {
            background-color: rgb(0, 105, 203);
            color: $color-aqua;
        }
    }

    .googleButton {
        display: flex;
        justify-content: center;
        align-items: center;
        margin-top: 20px;
        width: 300px;
        border: none;
        height: 40px;
        border-radius: 5px;
        gap: 10px;
        font-size: 17px;
        font-weight: 500;
        cursor: pointer;
        background-color: $color-white;
    }

    .googleButton:hover {
        background-color: rgb(0, 105, 203);
        color: $color-aqua;
    }

    p {
        font-size: 15px;

        a:hover {
            color: $color-aqua;
        }
    }
}

p {
    span {
        font-style: italic;
        color: blue;
        cursor: pointer;
        text-decoration: underline;
    }

    span:hover {
        color: rgb(167, 167, 255);
    }
}