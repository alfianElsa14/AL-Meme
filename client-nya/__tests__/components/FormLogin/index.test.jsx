import { fireEvent, render } from "@utils/testHelper";
import FormLogin from "@components/FormLogin";
import classes from "@components/FormLogin/style.module.scss"

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    useNavigate: () => mockNavigate,
}));

let wrapper;

beforeEach(() => {
    wrapper = render(<FormLogin />)
})

describe("Form login", () => {
    test('Should render correctly', () => {
        const { getByTestId } = wrapper;
        const FormLoginComponent = getByTestId('formContainer');
        expect(FormLoginComponent).toHaveClass(classes.formContainer)
        expect(FormLoginComponent).toBeInTheDocument();
    })

    test('Should render form login correctly', () => {
        const { getByTestId } = wrapper;
        const FormLoginComponent = getByTestId('formLogin');
        expect(FormLoginComponent).toHaveClass(classes.formLogin)
        expect(FormLoginComponent).toBeInTheDocument();
    })
    
    test('Should handle input email change', () => {
        const { getByTestId } = wrapper;
        const emailInput = getByTestId('email');
        fireEvent.change(emailInput, { target: { value : 'luak@gmail.com'}})
        expect(emailInput.value).toBe('luak@gmail.com')
    })

    test('Should handle input password change', () => {
        const { getByTestId } = wrapper;
        const passwordInput = getByTestId('password');
        fireEvent.change(passwordInput, { target: { value : '1234567'}})
        expect(passwordInput.value).toBe('1234567')
    })

    test('Should render button google login correctly', () => {
        const { getByTestId } = wrapper;
        const googleButton = getByTestId('google');
        expect(googleButton).toHaveClass(classes.googleButton)
        expect(googleButton).toBeInTheDocument();
    })

    test('Should call navigate when text clicked', () => {
        const { getByTestId } = wrapper;
        const text = getByTestId('toRegister');
        fireEvent.click(text);
        expect(mockNavigate).toHaveBeenCalledWith(`/register`);
    })

    test('Should match with snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})