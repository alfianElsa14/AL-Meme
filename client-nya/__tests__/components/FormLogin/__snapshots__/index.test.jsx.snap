// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form login Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="formContainer"
        data-testid="formContainer"
      >
        <h3>
          Masuk
        </h3>
        <div
          class="form"
        >
          <form
            action=""
            class="formLogin"
            data-testid="formLogin"
          >
            <label
              for=""
            >
              Email :
            </label>
            <input
              data-testid="email"
              type="email"
              value=""
            />
            <label
              for=""
            >
              Password :
            </label>
            <input
              data-testid="password"
              type="password"
              value=""
            />
            <button
              type="submit"
            >
              Masuk
            </button>
          </form>
          <button
            class="googleButton"
            data-testid="google"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="GoogleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.545,10.239v3.821h5.445c-0.712,2.315-2.647,3.972-5.445,3.972c-3.332,0-6.033-2.701-6.033-6.032s2.701-6.032,6.033-6.032c1.498,0,2.866,0.549,3.921,1.453l2.814-2.814C17.503,2.988,15.139,2,12.545,2C7.021,2,2.543,6.477,2.543,12s4.478,10,10.002,10c8.396,0,10.249-7.85,9.426-11.748L12.545,10.239z"
              />
            </svg>
            Log in dengan Google
          </button>
        </div>
        <p>
          Belum punya akun? 
          click 
          <span
            data-testid="toRegister"
          >
            Disini
          </span>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="formContainer"
      data-testid="formContainer"
    >
      <h3>
        Masuk
      </h3>
      <div
        class="form"
      >
        <form
          action=""
          class="formLogin"
          data-testid="formLogin"
        >
          <label
            for=""
          >
            Email :
          </label>
          <input
            data-testid="email"
            type="email"
            value=""
          />
          <label
            for=""
          >
            Password :
          </label>
          <input
            data-testid="password"
            type="password"
            value=""
          />
          <button
            type="submit"
          >
            Masuk
          </button>
        </form>
        <button
          class="googleButton"
          data-testid="google"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="GoogleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.545,10.239v3.821h5.445c-0.712,2.315-2.647,3.972-5.445,3.972c-3.332,0-6.033-2.701-6.033-6.032s2.701-6.032,6.033-6.032c1.498,0,2.866,0.549,3.921,1.453l2.814-2.814C17.503,2.988,15.139,2,12.545,2C7.021,2,2.543,6.477,2.543,12s4.478,10,10.002,10c8.396,0,10.249-7.85,9.426-11.748L12.545,10.239z"
            />
          </svg>
          Log in dengan Google
        </button>
      </div>
      <p>
        Belum punya akun? 
        click 
        <span
          data-testid="toRegister"
        >
          Disini
        </span>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
